module/dice/basic-roll.mjs

```diff
@@ -180,4 +180,100 @@ export default class BasicRoll extends Roll {
       return msg.toObject();
     }
   }
+
+  /* -------------------------------------------- */
+  /*  Evaluate Methods                            */
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  async evaluate({minimize=false, maximize=false, ...options}={}) {
+    if ( !this._evaluated ) {
+      if ( maximize ) this.replaceDieTermsMaximum(false);
+      else if ( minimize ) this.replaceDieTermsMaximum(true);
+    }
+    return super.evaluate(options);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  evaluateSync({minimize=false, maximize=false, ...options}={}) {
+    if ( !this._evaluated ) {
+      if ( maximize ) this.replaceDieTermsMaximum(false);
+      else if ( minimize ) this.replaceDieTermsMaximum(true);
+    }
+    return super.evaluateSync(options);
+  }
+
+  /* -------------------------------------------- */
+  /*  Maximize/Minimize Methods                   */
+  /* -------------------------------------------- */
+
+  /**
+   * Replaces all dice terms with their maximum/minimum value.
+   *
+   * @param {boolean} minimum                Option to calculate the minimum value.
+   * @returns {RollTerm[]}                   Returns the array of RollTerms.
+   */
+
+  replaceDieTermsMaximum(minimum = false) {
+    this.terms = this.terms.map(term => {
+      if (term instanceof DiceTerm) {
+        return new NumericTerm({ number: this.constructor.calcTermMaximum(term, minimum), options: term.options });
+      }
+      return term;
+    });
+    return this.terms;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Gets die information from passed die and calculates the maximum value that could be rolled.
+   *
+   * @param {DiceTerm} die                   DiceTerm to get the maximum/minimum value.
+   * @param {boolean} minimum                Option to calculate the minimum value.
+   * @returns {number}                       Maximum/Minimum value that could be rolled as an integer.
+   */
+
+  static calcTermMaximum(die, minimum = false) {
+    let face = minimum ? 1 : die.faces;
+    let number = die.number;
+    const currentModifiers = foundry.utils.deepClone(die.modifiers);
+    const validModifiers = {
+      k: "keep",
+      kh: "keep",
+      kl: "keep",
+      d: "drop",
+      dh: "drop",
+      dl: "drop",
+      max: "maximum",
+      min: "minimum"
+    };
+
+    for ( let modifier of currentModifiers ) {
+      const rgx = /([Mm][AaIi][XxNn]|[KkDd][HhLl]?)([\d]+)?/i;
```

Comment:
Since we are using the `i` flag, we do not need to include both upper- and lower-case letters in the regex. The `[\d]` character set is redundant also.

```suggestion
      const rgx = /(m[ai][xn]|[kd][hl]?)(\d+)?/i;
```